### PR TITLE
CCM governance recommendations and overview page

### PR DESCRIPTION
**Pull request by charanya-j**
Added two new markdown files to cover **Governance recommendations** and one for the **overview** page.
Updated **Set up cost visibility for AWS** with permissions for governance

- https://6462406c30002a20219fa209--harness-developer.netlify.app/docs/cloud-cost-management/use-ccm-cost-governance/asset-governance/gov-overview

- https://6462406c30002a20219fa209--harness-developer.netlify.app/docs/cloud-cost-management/use-ccm-cost-optimization/ccm-recommendations/governance

- https://6462406c30002a20219fa209--harness-developer.netlify.app/docs/cloud-cost-management/getting-started-ccm/set-up-cloud-cost-management/set-up-cost-visibility-for-aws#cloud-asset-governance

- https://6462406c30002a20219fa209--harness-developer.netlify.app/docs/cloud-cost-management/use-ccm-cost-optimization/optimize-cloud-costs-with-intelligent-cloud-auto-stopping-rules/load-balancer/create-load-balancer-aws - @riyasyash, this one sneaked in from another branch, but anyway needs to be published.

### CODE REVIEW

File: docs/cloud-cost-management/5-use-ccm-cost-governance/asset-governance/gov-overview.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+# Cloud asset governance overview page
+
+The **Overview** page offers a consolidated summary of your organization's governance rules, enforcements, and cost savings resulting from rule executions. This provides a quick and easy way to monitor your cloud spend and ensure compliance with your governance policies. Additionally, you can access valuable recommendations for optimizing resource utilization, allowing you to achieve even greater cost savings. 
+
+  <docimage path={require('./static/gov-overview.png')} width="50%" height="50%" title="Click to view full size image" />
+
+  - **All time realized savings** — The cost savings realized in the last one month. 
+  
+    You can view a visual representation of your savings trends through the **Savings Trend** graph, which allows you to hover over individual data points to view the corresponding cost savings for a specific day. This feature provides you with a granular level of insight into your cloud spending, enabling you to identify any trends or anomalies and make informed decisions to optimize your resource utilization and maximize cost savings.
```
Comment (Rohit-Kaliki) [inline docs/cloud-cost-management/5-use-ccm-cost-governance/asset-governance/gov-overview.md:15]: This is purely offering savings from CAG, this line doesn't make sense? 

We should also add a section on how the cost savings are calculated, in what scenarios etc. 
If we delete a EBS volume on 3rd May, we show costs saved on 3rd is 40$ (This 40$ is cost of running an EBS Volume for the last 30 days which is collected as savings) 

This feature provides you with a granular level of insight into your cloud spending, enabling you to identify any trends or anomalies and make informed decisions to optimize your resource utilization and maximize cost savings.

File: docs/cloud-cost-management/4-use-ccm-cost-optimization/1-ccm-recommendations/governance.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+   <docimage path={require('./static/governance.png')} width="50%" height="50%" title="Click to view full size image" />
+
+The following resource types can be utilized optimally by creating governance rules to rightsize the resources:
+
+- EC2 instances
+
+- RDS instances 
+
+- DynamoDB Tables
```
Comment (Rohit-Kaliki) [inline docs/cloud-cost-management/4-use-ccm-cost-optimization/1-ccm-recommendations/governance.md:21]: We don't have support for Dynamo, Lambda, ELB yet

File: docs/cloud-cost-management/4-use-ccm-cost-optimization/1-ccm-recommendations/governance.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+1. On the **Recommendations** page, select the recommendation row that you want to enforce. The following page appears.
+
+    <docimage path={require('./static/enforce-governance-recom.png')} width="50%" height="50%" title="Click to view full size image" />
+
+    * **Potential Monthly Savings** — Monthly cost savings that can be realized if the recommendations are applied.
+    * **Total Accounts** — The total number of AWS accounts on which the rules can be enforced.
+    * **Total Resources** — The total number of resources that would be impacted.
+2. Select the AWS account to which you want to apply the recommendation.
+3. Select Enforce.
+4. Enter a name for the Rule enforcement.
```
Comment (Rohit-Kaliki) [inline docs/cloud-cost-management/4-use-ccm-cost-optimization/1-ccm-recommendations/governance.md:77]: We should also highlight the fact that on the first run, it essentially rightsizes/ decommisions active opp. but the enforcement will ensure savings are ongoing.

**Review by Rohit-Kaliki — CHANGES_REQUESTED** [review]

File: docs/cloud-cost-management/5-use-ccm-cost-governance/asset-governance/gov-overview.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+# sidebar_position: 2
+---
+
+# Cloud asset governance overview page
+
+The **Overview** page offers a consolidated summary of your organization's governance rules, enforcements, and cost savings resulting from rule executions. This provides a quick and easy way to monitor your cloud spend and ensure compliance with your governance policies. Additionally, you can access valuable recommendations for optimizing resource utilization, allowing you to achieve even greater cost savings. 
+
+  <docimage path={require('./static/gov-overview.png')} width="50%" height="50%" title="Click to view full size image" />
+
+  - **All time realized savings** — The cost savings realized in the last one month.
```
[the remark this answers is not in the document]
Comment (nikunj-badjatya) [reply docs/cloud-cost-management/5-use-ccm-cost-governance/asset-governance/gov-overview.md:13]: Yes confirmed.

File: docs/cloud-cost-management/2-getting-started-ccm/4-set-up-cloud-cost-management/set-up-cost-visibility-for-aws.md [hunk 5]
```diff
@@ -408,6 +409,51 @@ This feature provides visibility into your EC2, EBS volumes, and ECS costs. The
...
+            "Action": [
+                "ec2:Describe*",
+                "ec2:DeleteSnapshot",
+                "ec2:DeleteVolume",
+                "ec2:Get*",
+                "ec2:ListImagesInRecycleBin",
+                "ec2:ListSnapshotsInRecycleBin",
+                "elasticbeanstalk:Check*",
+                "elasticbeanstalk:Describe*",
+                "elasticbeanstalk:List*",
```
Comment (nikunj-badjatya) [inline docs/cloud-cost-management/2-getting-started-ccm/4-set-up-cloud-cost-management/set-up-cost-visibility-for-aws.md:429]: We will need to update these. https://github.com/harness/harness-core/pull/47851/files once this is merged we can take the permissions from here.

File: docs/cloud-cost-management/5-use-ccm-cost-governance/asset-governance/gov-overview.md (hunk 1 above)
Comment (charanya-j) [reply docs/cloud-cost-management/5-use-ccm-cost-governance/asset-governance/gov-overview.md:15]: realized cost savings
enforcement -2
ec2 old instance and ec2 low usage
the same VM (instance or resource) can show up in both rules
these rules are independent and the cost is calculated independently for each execution. We need to note that there might be an overlap.

**Review by riyasyash — APPROVED** [review]

**Comment by riyasyash** [comment]
@charanya-j Reviewed for lambda code update section doc

**Comment by charanya-j** [comment]
Updated comments and created another pull request and merged 
https://github.com/harness/developer-hub/pull/1815

**Pull request closed without merging**